treeopt/treeOpt.py

```diff
@@ -0,0 +1,473 @@
+        be visualized, the value of the keyword is checked and depending on the
+        keyword a different plot is generated
+        :param keyword: Keyword describing the problem-type (default "extern",
+        other alternative "benchmark")
+        :type keyword: String
+        :return: Nothing
+        :rtype: None
+
+        """
+
+        self.sim_keyword = keyword
+
+    def set_sm_method(self, method):
+        """
+        Sets the Method that is used to approximate the system responce in the
+        design space
+        :param method: one of the functions in modules/metamodell.py
+        :type method: python function
+        :return: Nothing
+        :rtype: None
+
+        """
+
+        self.smMethod = method
+
+    def set_accuracy_criterion(self, method):
+        """
+        Sets the method that is used to define the termination condition of the
+        optimization workflow
+        :param method: one of the functions in modules/accuracy
+        :type method: Python Function
+        :return: Nothing
+        :rtype: None
+
+        """
+
+        self.accuracyCriterion = method
+
+    def simulate_problem(self, x):
+        """
+        Executes the problem which is to be analyzed. If static Variables where
+        defined as additional arguments for the problem, these arguments are
+        added to the function call.
+        :param x: Point in the design space
+        :type x: Numpy-array
+        :return: System responce
+        :rtype: Numpy-array
+
+        """
+
+        if self.problem_args is None:
+            return self.problem(x)
+        else:
+            return self.problem(x, self.problem_args)
+
+    def start_optimization(self):
+        """
+        Function that starts the previosly parameterized adaptive optimization
+        loop
+        :return: Nothing
+        :rtype: None
+
+        """
+
+        self.x = self.samplingMethod(self.limits, self.numDOE)
+
+        self.y = self.simulate_problem(np.atleast_2d(self.x[0]))
+        for xi in self.x[1:]:
+            print(self.simulate_problem(np.atleast_2d(xi)))
+            self.append_y_data(self.simulate_problem(np.atleast_2d(xi)))
+
+        self.write_data(self.x, "DoeData")
+        self.write_data(self.y, "DoeResponce")
+
+        self.optGoal = False
+
+        self.ite = 0
+        self.maxite = 5
+        self.all_nx_var = []
+        self.all_success = []
+
+        while self.optGoal is False:
+
+            self.sm = self.smMethod(self.x, self.y)
+
+            self.nX = optimize.get_lowest_variance(self.sm, self.limits)
+
+            self.append_x_data(self.nX)
+            self.append_y_data(self.simulate_problem(np.atleast_2d(self.nX)))
+
+            #Writes the Data in a file.
+            self.write_data(self.x, "DoeData")
+            self.write_data(self.y, "DoeResponce")
+
+            self.ite += 1
+
+            if self.ite == self.maxite:
+                self.optGoal = True
```

Comment:
Does not need to be fixed as part of this PR: `self.optGoald = True` implies that you achieved your optimization goal, once you reach the maximum number of iterations. Is there no other way to check the quality of the current result?